src/sql/workbench/contrib/queryplan2/browser/queryPlan.ts

```diff
@@ -80,486 +107,317 @@ export class QueryPlan2View implements IPanelView {
-
-			remoteIndexScan: imageBasePath + 'remote_index_scan.png',
-
-			remoteIndexSeek: imageBasePath + 'remote_index_seek.png',
-
-			remoteInsert: imageBasePath + 'remote_insert.png',
-
-			remoteQuery: imageBasePath + 'remote_query.png',
-
-			remoteScan: imageBasePath + 'remote_scan.png',
-
-			remoteUpdate: imageBasePath + 'remote_update.png',
-
-			ridLookup: imageBasePath + 'rid_lookup.png',
-
-			rowCountSpool: imageBasePath + 'row_count_spool.png',
-
-			segment: imageBasePath + 'segment.png',
-
-			sequence: imageBasePath + 'sequence.png',
-
-			sequenceProject: imageBasePath + 'sequence_project.png',
-
-			sort: imageBasePath + 'sort.png',
-
-			split: imageBasePath + 'split.png',
-
-			streamAggregate: imageBasePath + 'stream_aggregate.png',
-
-			switchStatement: imageBasePath + 'switch.png',
-
-			tableValuedFunction: imageBasePath + 'table_valued_function.png',
-
-			tableDelete: imageBasePath + 'table_delete.png',
-
-			tableInsert: imageBasePath + 'table_insert.png',
-
-			tableScan: imageBasePath + 'table_scan.png',
-
-			tableSpool: imageBasePath + 'table_spool.png',
-
-			tableUpdate: imageBasePath + 'table_update.png',
-
-			tableMerge: imageBasePath + 'table_merge.png',
-
-			tfp: imageBasePath + 'predict.png',
-
-			top: imageBasePath + 'top.png',
-
-			udx: imageBasePath + 'udx.png',
-
-			batchHashTableBuild: imageBasePath + 'batch_hash_table_build.png',
-
-			windowSpool: imageBasePath + 'table_spool.png',
-
-			windowAggregate: imageBasePath + 'window_aggregate.png',
+		this._actionBar.pushAction(actions, { icon: true, label: false });
 
-			// cursor operators
-			fetchQuery: imageBasePath + 'fetch_query.png',
 
-			populateQuery: imageBasePath + 'population_query.png',
-
-			refreshQuery: imageBasePath + 'refresh_query.png',
-
-			// shiloh operators
-			result: imageBasePath + 'result.png',
-
-			aggregate: imageBasePath + 'aggregate.png',
-
-			assign: imageBasePath + 'assign.png',
-
-			arithmeticExpression: imageBasePath + 'arithmetic_expression.png',
-
-			bookmarkLookup: imageBasePath + 'bookmark_lookup.png',
-
-			convert: imageBasePath + 'convert.png',
-
-			declare: imageBasePath + 'declare.png',
-
-			deleteOperator: imageBasePath + 'delete.png',
-
-			dynamic: imageBasePath + 'dynamic.png',
-
-			hashMatchRoot: imageBasePath + 'hash_match_root.png',
-
-			hashMatchTeam: imageBasePath + 'hash_match_team.png',
-
-			ifOperator: imageBasePath + 'if.png',
-
-			insert: imageBasePath + 'insert.png',
-
-			intrinsic: imageBasePath + 'intrinsic.png',
-
-			keyset: imageBasePath + 'keyset.png',
+	}
 
-			locate: imageBasePath + 'locate.png',
+	getHorizontalSashTop(sash: Sash): number {
+		return 0;
+	}
+	getHorizontalSashLeft?(sash: Sash): number {
+		return 0;
+	}
+	getHorizontalSashWidth?(sash: Sash): number {
+		return this._container.clientWidth;
+	}
 
-			populationQuery: imageBasePath + 'population_query.png',
+	private populate(node: azdata.ExecutionPlanNode, diagramNode: any): any {
```

Comment:
These should have a type instead of any

Reply:
Hm, that's too bad. A follow up to add a typings file would be great. You could look at https://www.npmjs.com/package/@typed-mxgraph/typed-mxgraph and see if that's accurate enough to use.